src/server/game/Spells/Spell.cpp

```diff
@@ -5498,7 +5499,7 @@ SpellCastResult Spell::CheckCast(bool strict, uint32* param1 /*= nullptr*/, uint
                     else if (m_preGeneratedPath->IsInvalidDestinationZ(target)) // Check position z, if in a straight line
                             return SPELL_FAILED_NOPATH;
 
-                    m_preGeneratedPath->ReducePathLenghtByDist(objSize); // move back
+                    m_preGeneratedPath->ShortenPathUntilDist(PositionToVector3(target), objSize); // move back
```

Comment:
just pass x,y,z separately?

Reply:
sure - but in how many places do we have to do that before it's unneeded code duplication? helpers are lightweight

Reply:
you know what I'm hinting at here (include hater)

Reply:
the only thing new is the `.hpp` - we already had `Vector3` (from `PathGenerator` itself)